components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacet.java

```diff
@@ -24,19 +24,23 @@
 public interface SearchFacet
     extends Facet
 {
+  /**
+   * Returns a unique identifier for the index entry of this component.
+   */
+  SearchItemId identifier(Component component);
 
   /**
    * Index component metadata.
    *
    * @param component to be indexed
    */
-  void put(Component component);
+  SearchItemId put(Component component);
```

Comment:
Minor: if this was `void put(SearchItemId identifier)` then it would match `delete` operation, otherwise in one case you need the component and in the other case its identifier. (this is mainly stylistic, downside of making this change is that callers would then need to transform component->identifier themselves.)